ext/native.go

```diff
@@ -711,6 +711,9 @@ func getFieldValue(adapter types.Adapter, refField reflect.Value) any {
 	if refField.IsZero() {
 		switch refField.Kind() {
 		case reflect.Array, reflect.Slice:
+			if refField.Type().Elem() == reflect.TypeOf(byte(0)) {
+				return refField.Interface()
+			}
 			return types.NewDynamicList(adapter, []ref.Val{})
```

Comment:
@TristonianJones thinking about the logic here, what do you think about also returning the interface value of zero types (besides pointers probably). The way I understand it, shouldn't then further down the chain native type be converted via the type adapter?

Reply:
What you suggest seems like the right choice to me since something elsewhere in the stack should do the proper type conversion using the adapter. This is likely an oversight on my part ... worth trying for sure!

Reply:
It looks like this change works in some cases, but causes breakages in others. My guess is that nested type accesses need to be proxied through the proper `adapter` instance, so only scalar types should return `.Interface()` and the complex types should go through the previous logic related to lists, maps, and structs.

Reply:
Ah my bad! Looks like I didn't run the tests again after my change! The problem has been fixed.